src/features/parser/root/parseRootPageContent.ts

```diff
@@ -0,0 +1,87 @@
+export interface ModuleContent {
+    code: string;
+    url: string;
+}
+
+/**
+ * Main function that orchestrates the parsing of the root page.
+ * It delegates the extraction of taught students' links and modules to other functions.
+ * @param {HTMLElement} content - The content of the root page to parse.
+ * @returns {object} An object containing arrays of ModuleContent for modules and taught students.
+ */
+export function parseRootPage(content: HTMLElement): {
+    modules: ModuleContent[],
+    taught_students: ModuleContent[]
+} {
+    const taught_students = extractTaughtStudentsLinks(content);
+    const modules = extractModulesLinks(content);
+
+    return {
+        modules: modules,
+        taught_students: taught_students
+    };
+}
+
+/**
+ * Extracts the links related to taught students from the specific <p> element.
+ * @param {HTMLElement} content - The content of the page to parse.
+ * @returns {ModuleContent[]} An array of ModuleContent objects representing the links and their labels.
+ */
+export function extractTaughtStudentsLinks(content: HTMLElement): ModuleContent[] {
+    const modules: ModuleContent[] = [];
+
+    const linkContainer = content.querySelector('p a[href*="studres.cs.st-andrews.ac.uk/Teaching/"]')?.parentElement;
+
+    if (linkContainer) {
+        const links = linkContainer.querySelectorAll('a');
+
+        links.forEach(link => {
+            const moduleCode = link.textContent?.trim() || "";
+            const moduleUrl = link.getAttribute('href') || "";
```

Comment:
moduleUrl returns link to studres page without '/' which changes appearence of the page. Might need to add '/' in the end, but don't know how

Reply:
## Module code

use function `extractUrlSegments(url)` and then first item of array will be module code

```ts
const urlSegments = extractUrlSegments(someurl)

const moduleCode = urlSegments[0]
```

----

## Potential link fix

instead of using `link.getAttribute('href)` use just `link.href`. 

Reason for this is that 1st one returns actual value of href (smth like '/some/path') and the other one returns full URL (https://some.url/path).

But there might be bigger issue with Providers not working correctly(theme and smooth nav not working)

Reply:
![image](https://github.com/user-attachments/assets/85ef392c-d4c5-4f3f-b04d-4455646b4c32)

We fixed link, but as you mentioned, there are probably a bigger issue with Providers.